students/DrewSmith/lessons/lesson09/assignment/database.py

```diff
@@ -0,0 +1,150 @@
+'''
+HP Norton inventory and customer information database functions
+'''
+
+
+import csv
+import pathlib
+import pymongo
+import logging
+from pymongo import MongoClient
+import assignment.config as config
+
+
+class MongoDBConnection(object):
+    """MongoDB Connection"""
+    def __init__(self, host='127.0.0.1', port=27017):
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        ''' Open connection '''
+        if self.connection is None:
+            try:
+                self.connection = MongoClient(self.host, self.port)
+            except pymongo.errors.InvalidURI as error:
+                logging.error(f"MongoDB InvalidURI: {error}")
+                raise
+            except pymongo.errors.ConfigurationError as error:
+                logging.error(f"MongoDB configurationError: {error}")
+                raise
+            except pymongo.errors.ConnectionFailure as error:
+                logging.error(f"MongoDB ConnectionFailure: {error}")
+                raise            
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        ''' Close connection '''
+        if self.connection is not None:
+            self.connection.close()
+        return False
```

Comment:
Cool. In the end, I'm not sure this is worth it, as in I'm not sure it is really adding much value. You could accomplish pretty much the same thing, by adding a logging statement in __exit__ that logged anything if there was an exception. I guess it would be broader in the exceptions it caught, but that would probably not be a bad thing.
But it was a vague assignment, and you got some good practice working with context managers. :)